include/aspect/particle/property/viscoplastic_strain_invariants.h

```diff
@@ -101,6 +101,7 @@ namespace aspect
            * but it is currently not used in a threaded context.
            */
           mutable MaterialModel::MaterialModelInputs<dim> material_inputs;
+          mutable MaterialModel::MaterialModelOutputs<dim> material_outputs;
```

Comment:
Add the following here for the speedup we discussed:

```suggestion
          mutable MaterialModel::MaterialModelOutputs<dim> material_outputs;
          
          /**
           * Store the indices of relevant compositional fields to avoid repeated lookups.
           */
          unsigned int plastic_strain_index;
          unsigned int viscous_strain_index;
          unsigned int total_strain_index;
```